contents/security/passwordStorage.md

```diff
@@ -0,0 +1,126 @@
+
+<popover id="pop:encrypt" title="Encryption refers to the process of changing a message such that it becomes essentially random text." placement="top">
+  <div slot="content">
+ The opposite process is called decryption, which converts the text back into the original message.
+  </div>
+</popover>
+
+<popover id="pop:oneway" title="" placement="top">
+<div slot="content">
+	A one way function is a function that is easy to compute the result of, but whose results are difficult to reverse back to the original input.
+</div>
+</popover>
+
+## What is hashing?
+
+Hashing is the act of transforming a set of data into another set of data that is impossible to reverse. As an example, suppose you wanted to hash [the entire play of Romeo and Juliet](http://shakespeare.mit.edu/romeo_juliet/full.html). If you performed hashing using SHA1 on the entire contents of that file, excluding the header, you'll end up hashing the entire 6048 lines of that file into `7A52CDA9F04051B0D8263AA61A42531DD27FCE92`. 
+
+Obviously, it is impossible to condense all 6048 lines of text into just those 40 characters. This means that is is impossible to recreate the entire play of Romeo and Juliet from just the hash result. Thus, we say that a hashing function is a one-way function, as it can only be performed in a one direction, and it would be impossible to reverse the process. 
+
+Additionally, due to how hashing algorithms work, even a small change in the file will result in a completely different hash. For example, if I were to change the first line from `ACT I` to `ACT IX`, it would change the hash to `D180AE51CEEFA8EF6D01F7FD54528AC21659433E`, completely different from the previous hash. This means that even very similar passwords that vary only by capitalization or a letter would still result in different <trigger for="pop:hashes">hashes</trigger>, so attackers can't compare the hash to try and figure out which users have similar passwords.
+
+<popover id="pop:hashes" title="" placement="top">
+<div slot="content">
+  Hashes are the hash result of a hashing function.
+</div>
+</popover>
+
+Some examples of cryptographic hashing algorithms are [MD5](https://www.quora.com/How-does-the-MD5-algorithm-work) and [SHA1](https://deadhacker.com/2006/02/21/sha-1-illustrated/). However, when you perform hashing for passwords, you should use password hashing algorithms (such as [Argon2](https://github.com/P-H-C/phc-winner-argon2), [SCrypt](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.scrypt.html) and [bcrypt](https://security.stackexchange.com/questions/4781/do-any-security-experts-recommend-bcrypt-for-password-storage)). These password hashing algorithms are designed to be slow to prevent brute force attacks, unlike cryptographic hashing algorithms which are built for speed. Despite that, if you plan on learning more about hashing algorithms, I recommend starting with MD5 and SHA1, as they are easier algorithms to learn about.
```

Comment:
It's best to use 'we' rather than 'I' as the article already involves more than one person. In the future, there might be more than one author too.